include/nbl/video/utilities/CScanner.h

```diff
@@ -167,18 +165,18 @@ class CScanner final : public core::IReferenceCounted
 		};
 		enum E_OPERATOR : uint8_t
 		{
-			EO_AND = _NBL_GLSL_SCAN_OP_AND_,
-			EO_XOR = _NBL_GLSL_SCAN_OP_XOR_,
-			EO_OR = _NBL_GLSL_SCAN_OP_OR_,
-			EO_ADD = _NBL_GLSL_SCAN_OP_ADD_,
-			EO_MUL = _NBL_GLSL_SCAN_OP_MUL_,
-			EO_MIN = _NBL_GLSL_SCAN_OP_MIN_,
-			EO_MAX = _NBL_GLSL_SCAN_OP_MAX_,
-			EO_COUNT = _NBL_GLSL_SCAN_OP_COUNT_
+			EO_AND = 0u,
+			EO_XOR,
+			EO_OR,
+			EO_ADD,
+			EO_MUL,
+			EO_MIN,
+			EO_MAX,
+			EO_COUNT
 		};
 
 		// This struct is only for managing where to store intermediate results of the scans
-		struct Parameters : nbl_glsl_scan_Parameters_t // this struct and its methods are also available in GLSL so you can launch indirect dispatches
+		struct Parameters : nbl::hlsl::scan::Parameters_t // this struct and its methods are also available in GLSL so you can launch indirect dispatches
 		{
 			static inline constexpr uint32_t MaxScanLevels = NBL_BUILTIN_MAX_SCAN_LEVELS;
 
```

Comment:
you can probably move the entire struct to HLSL since HLSL allows for methods now :D

might need to port `roundUp` and `exclusive_scan` to HLSL `algorithm.hlsl`